RocksDB exception on restart

### Description
We've got a user reporting a rocksdb issue after shutting down teku and restarting:
```
08:09:15.531 INFO  - Teku version: teku/v0.12.4-dev-1bbea6b0/linux-x86_64/-ubuntu-openjdk64bitservervm-java-11
08:09:15.828 INFO  - Loading 1 validator keys...
08:09:19.157 INFO  - Loaded 1 Validators: b5d7418
08:09:19.476 INFO  - Using data path: /home/user/.local/share/teku/data
Teku failed to start.

Caused by: tech.pegasys.teku.storage.server.DatabaseStorageException: Failed to open database at path: ~/.local/share/teku/data/db

Caused by: org.rocksdb.RocksDBException: SST file is ahead of WALs
```

The user reports that they used `ctrl-c` to stop teku before restarting.  

Problematic data directory is available [here](https://www.dropbox.com/s/drly52998t9sx8x/data.zip).

### Versions
* Software version: v0.12.4-dev-1bbea6b0
* Java version: java 11

--- a/gradle/versions.gradle
+++ b/gradle/versions.gradle
@@ -122,7 +122,7 @@ dependencyManagement {
     dependency 'org.openjdk.jmh:jmh-generator-annprocess:1.23'
     dependency 'org.miracl.milagro.amcl:milagro-crypto-java:0.4.0'
     dependency 'org.quartz-scheduler:quartz:2.3.2'
-    dependency 'org.rocksdb:rocksdbjni:6.10.2'
+    dependency 'org.rocksdb:rocksdbjni:6.11.4'
 
     dependencySet(group: "org.web3j", version: "4.6.2") {
       entry 'core'

--- a/storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/core/RocksDbInstance.java
+++ b/storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/core/RocksDbInstance.java
@@ -160,6 +160,7 @@ public class RocksDbInstance implements RocksDbAccessor {
       for (Transaction openTransaction : openTransactions) {
         openTransaction.closeViaDatabase();
       }
+      db.syncWal();
       for (final AutoCloseable resource : resources) {
         resource.close();
       }

